traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go

```diff
@@ -425,6 +446,30 @@ func GetCreateHandler(w http.ResponseWriter, r *http.Request) {
 	api.WriteResp(w, r, tc.DeliveryServiceServers{payload.ServerNames, payload.XmlId})
 }
 
+// ValidateServerCapabilities checks that the delivery service's requirements are met by each server to be assigned.
+func ValidateServerCapabilities(dsID int, serverNames []string, tx *sql.Tx) error {
```

Comment:
this need to differentiate between storage errors from `GetDSRequiredCapabilitiesFromID ` and not meeting the delivery service req capabilities.

This is because the return back to the user differs. The first should be an internal server error and the latter needs to be bad request.

Right now getting internal server error on both.